engine/runtime/src/main/java/org/enso/interpreter/runtime/data/text/Text.java

```diff
@@ -74,8 +74,17 @@ String toDisplayString(
       boolean allowSideEffects,
       @Cached("build()") @Cached.Shared("strings") ToJavaStringNode toJavaStringNode) {
     String str = toJavaStringNode.execute(this);
-    String replaced = str.replace("\"", "\\\"");
-    return "\"" + replaced + "\"";
+    String replaced =
+        str.replace("'", "\\'")
+            .replace("\n", "\\n")
+            .replace("\t", "\\t")
+            .replace("\u0007", "\\a")
+            .replace("\u0008", "\\b")
+            .replace("\u000c", "\\f")
+            .replace("\r", "\\r")
+            .replace("\u000B", "\\v")
+            .replace("\u001B", "\\e");
+    return "'" + replaced + "'";
```

Comment:
I feel like we probably need a more extensible way to do this. Can you add a `TODO` comment for it? Also, can you add some printing tests for this?